projects/packages/post-list/src/class-post-list.php

```diff
@@ -153,6 +153,53 @@ public function add_filters_and_actions_for_screen( $current_screen ) {
 
 		$this->maybe_customize_columns( $current_screen->post_type );
 		$this->maybe_add_share_action( $current_screen->post_type );
+		$this->maybe_add_copy_link_action( $current_screen->post_type );
+	}
+
+	/**
+	 * Checks the current post type and adds the Copy link post
+	 * action if it is appropriate to do so.
+	 *
+	 * @param string $post_type The post type associated with the current request.
+	 */
+	public function maybe_add_copy_link_action( $post_type ) {
+		if ( is_post_type_viewable( $post_type ) ) {
```

Comment:
We're intentionally adding the link even if the status is non-vieweable, right? (draft for example)

Reply:
No. We add the script here based on `post type`. We later check on `is_post_publicly_viewable` for each entity. I thought about adding only `is_post_status_viewable` there but it seemed maybe clearer for readability. I'm still torned about this 😄

Reply:
Well, admins might still want to copy a link to their draft post, right?